### PR TITLE
Rya 81

### PR DESCRIPTION
**Pull request by kchilton2**

### CODE REVIEW

File: common/rya.api/src/main/java/mvm/rya/api/instance/RyaDetails.java [hunk 1]
```diff
@@ -0,0 +1,908 @@
...
+
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import javax.annotation.concurrent.Immutable;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Details about how a Rya instance's state.
```
Comment (isper3at) [inline common/rya.api/src/main/java/mvm/rya/api/instance/RyaDetails.java]: wording is funny

File: common/rya.api/src/main/java/mvm/rya/api/instance/RyaDetails.java [hunk 2]
```diff
@@ -0,0 +1,908 @@
...
+            return false;
+        }
+    }
+
+    /**
+     * Details about a Rya instance's Join Selectivity statistics.
+     */
+    @Immutable
+    @ParametersAreNonnullByDefault
+    public static class JoinSelectivityDetails implements Serializable {
```
Comment (isper3at) [inline common/rya.api/src/main/java/mvm/rya/api/instance/RyaDetails.java]: any reason these aren't in their own files?

File: extras/indexing/src/main/java/mvm/rya/indexing/pcj/matching/PCJOptimizer.java [hunk 3]
```diff
@@ -76,273 +80,277 @@
...
+     * LeftJoin. Once it reaches this node, it gets the appropriate PCJMatcher
+     * from the {@link QuerySegmentPCJMatchVisitor} and uses this to match each
+     * of the PCJs to the {@link QuerySegment} starting with the Join, Filter,
+     * or LeftJoin. Once each PCJ has been compared for matching, the portion of
+     * the query starting with the Join, Filter, or LeftJoin is replaced by the
+     * {@link TupleExpr} returned by {@link PCJMatcher#getQuery()}.  This visitor
+     * then visits each of the nodes returned by {@link PCJMatcher#getUnmatchedArgs()}.
+     *
+     */
+    static class QuerySegmentPCJMatchVisitor extends
```
Comment (isper3at) [inline extras/indexing/src/main/java/mvm/rya/indexing/pcj/matching/PCJOptimizer.java]: word wrap is a little short

File: extras/indexing/src/main/java/mvm/rya/indexing/pcj/matching/PCJOptimizer.java [hunk 4]
```diff
@@ -76,273 +80,277 @@
...
+
+            for (TupleExpr tupleExpr : unmatched) {
+                tupleExpr.visit(this);
+            }
+        }
+
+    }
+
+    /**
+     *
```
Comment (isper3at) [inline extras/indexing/src/main/java/mvm/rya/indexing/pcj/matching/PCJOptimizer.java]: docs

File: extras/indexing/src/main/java/mvm/rya/indexing/pcj/matching/PCJOptimizer.java [hunk 5]
```diff
@@ -334,9 +334,11 @@ public void meet(Filter node) {
         } else {
             //if no tables are provided by user, get ids for rya instance id, create table name,
             //and associate table name with sparql
+            // TODO: storage.listPcjs() returns tablenames, not PCJ-IDs.  
+            //     See mvm.rya.indexing.external.accumulo.AccumuloPcjStorage.dropPcj(String)
```
Comment (isper3at) [inline extras/indexing/src/main/java/mvm/rya/indexing/pcj/matching/PCJOptimizer.java]: in caleb's commit there is a getPcjId()

File: extras/indexing/src/test/java/mvm/rya/indexing/accumulo/temporal/AccumuloTemporalIndexerTest.java [hunk 6]
```diff
@@ -21,9 +21,8 @@
 
 
 import static mvm.rya.api.resolver.RdfToRyaConversions.convertStatement;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
-
+import static org.junit.Assert.*;
```
Comment (isper3at) [inline extras/indexing/src/test/java/mvm/rya/indexing/accumulo/temporal/AccumuloTemporalIndexerTest.java]: dont use *

File: extras/indexing/src/test/java/mvm/rya/indexing/accumulo/temporal/AccumuloTemporalIndexerTest.java [hunk 7]
```diff
@@ -237,17 +219,15 @@ public static void tearDownAfterClass() throws Exception {
     @Before
     public void setUp() throws Exception {
         conf = new Configuration();
+        conf.set(RdfCloudTripleStoreConfiguration.CONF_TBL_PREFIX, "triplestore_");
         conf.setBoolean(ConfigUtils.USE_MOCK_INSTANCE, true);
-        conf.set(ConfigUtils.TEMPORAL_TABLENAME, uniquePerTestTemporalIndexTableName);
-        // This is from http://linkedevents.org/ontology
+        // The temporal predicates are from http://linkedevents.org/ontology
```
Comment (isper3at) [inline extras/indexing/src/test/java/mvm/rya/indexing/accumulo/temporal/AccumuloTemporalIndexerTest.java]: already linked in the class doc

File: extras/indexing/src/test/java/mvm/rya/indexing/external/tupleSet/AccumuloIndexSetColumnVisibilityTest.java [hunk 8]
```diff
@@ -60,6 +36,34 @@
 import com.google.common.collect.Sets;
 import com.google.common.io.Files;
 
+/*
```
Comment (isper3at) [inline extras/indexing/src/test/java/mvm/rya/indexing/external/tupleSet/AccumuloIndexSetColumnVisibilityTest.java]: thought we were putting these above package

**Comment by amihalik** [comment]
Merged.

**Pull request closed without merging**